applications/crossbar/priv/couchdb/schemas/call_recordings.json

```diff
@@ -106,6 +111,71 @@
             "name": "request",
             "type": "string"
         },
+        "soft_delete": {
```

Comment:
KAZOO already has a backend concept of soft-deletes. When a DELETE is processed the document is soft-deleted by adding `pvt_deleted` (and all the views account for this when returning the listing). If a document is deleted accidentally, it can be un-deleted by an admin by removing this key from the doc (provided the doc still exists). There is a periodic cleanup of soft-deleted documents.

As the backend doesn't use these fields, I don't think adding them to the schema makes sense. Your client can add them and KAZOO will happily store the fields for you. You can then issue the GET with the appropriate query string filter to ignore docs with `soft_delete` defined... again this is all client-side work and not something the backend needs to be responsible for.

Happy to hear your reasoning and discuss this further, of course!

Reply:
Problem is that the schema has "additionalProperties": false so Kazoo will not happilly store the fields unless we change it to true and we assumed it was set to false for a good reason

Reply:
The soft-deleted document will contain the pvt_request_id and the account/user info of who last touched the doc (in the DELETE case, who issued the DELETE). So this seems to duplicate that functionality anyway.

Reply:
Hi James

The idea behind this was that in the call recording app we have overrides that allow users  / admins to mark documents as deleted. When they do that we patch the recording/recordings collection and mark them as soft_deleted. The reason we didn't use the existing delete is that we didn't see a way to list the deleted documents via the api to allow an admin to restore the metadata without having to go into couch to restore. If you would prefer we approach this in a different way then please let me know.

Thanks
Brian